src/librustc_resolve/lib.rs

```diff
@@ -122,7 +122,7 @@ impl Ord for BindingError {
 
 enum ResolutionError<'a> {
     /// error E0401: can't use type parameters from outer function
-    TypeParametersFromOuterFunction,
+    TypeParametersFromOuterFunction(Option<&'a Span>),
```

Comment:
May I suggest, instead of passing the span, that you pass the `Def`? (See other comments)